replace hashicorp/go-multierror with errors.Join

thaJeztah opened the pull request:
- [x] depends on https://github.com/opencontainers/runtime-tools/pull/776

thaJeztah committed:
replace hashicorp/go-multierror with errors.Join

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Ah, right, this requires go1.20 as minimum, which perhaps by now may be fine?

kolyshkin commented:
@thaJeztah #776 is merged; time to rebase

thaJeztah commented on validate/validate.go:
```diff
@@ -144,21 +144,21 @@ func (v *Validator) CheckJSONSchema() (errs error) {
 	url, err := JSONSchemaURL(strings.TrimSuffix(v.spec.Version, "-dev"))
 	if err != nil {
-		errs = multierror.Append(errs, err)
+		errs = errors.Join(errs, err)
```
I need to check if I'm actually doing these right; I recently made a mistake with stdlib "multi-errors" because (IIRC) `errors.Join` is more an equivalent to `errors.Wrap`, and _nests_ a new error, not making it on the same "level" as other errors when using a loop.

kolyshkin commented:
With #777 we're switching to go 1.21 as a minimum so once merged we can merge this one as well.

thaJeztah commented:
Rebased after #777 was merged

kolyshkin reviewed (dismissed):
lgtm

tianon commented:
I think https://github.com/opencontainers/runtime-tools/graphs/contributors tells a pretty clear story about whether or not anyone is actively maintaining the code in this repository in 2025 (and frankly even as far back as 2019), and I'd honestly suggest that any project finding value from it should probably be thinking about either forking or adapting to some new library, if there's some code here that's still useful.

Is there a particular reason _you're_ looking at this, Seb?  Some transitive dependency pulling this into a project you maintain?

> oci-runtime-tool is a collection of tools for working with the OCI runtime specification.

Like honestly, this description is really `runc`, right?  I get that this repository does something slightly different, but to what end?  We had some lofty goals, but I don't think they were ever really realized.

thaJeztah committed:
validate: refactor multi-errors

Use a slightly more idiomatic approach, and collect errors before
joining them. This prevents deeply nested error trees.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
Updated, but looks like CI started failing because golangci-lint doesn't like go1.26; 😂 

```
run golangci-lint
  Running [/home/runner/golangci-lint-2.6.2-linux-amd64/golangci-lint config path] in [/home/runner/work/runtime-tools/runtime-tools] ...
  Running [/home/runner/golangci-lint-2.6.2-linux-amd64/golangci-lint config verify] in [/home/runner/work/runtime-tools/runtime-tools] ...
  Running [/home/runner/golangci-lint-2.6.2-linux-amd64/golangci-lint run] in [/home/runner/work/runtime-tools/runtime-tools] ...
  panic: file requires newer Go version go1.26 (application built with go1.25) [recovered, repanicked]
```

thaJeztah commented:
@giuseppe @kolyshkin @cyphar rebased after golangci-lint was updated; PTAL 🤗

giuseppe reviewed (approved):
LGTM
